domain/src/main/kotlin/com/krzykrucz/elesson/currentlesson/shared/Base.kt

```diff
@@ -69,7 +69,18 @@ class NonEmptySet<T> private constructor(private val elements: Set<T>) : Set<T>
 
         fun <T> create(set: Set<T>): NonEmptySet<T>? = NonEmptySet(set)
     }
+}
+
+class NonEmptyList<T> private constructor(private val elements: List<T>) : List<T> by elements {
```

Comment:
Arrow has a `NonEmptyList` maybe we could use that?

Reply:
very good idea! I didn't use this NonEmptyList at last, however

I will remove it from here